### PR TITLE
(PUP-10040) Use http client for cert related requests

### PR DESCRIPTION
**Pull request by joshcooper**
Change the `puppet ssl` application and the ssl state machine to use the http client, and deprecate the `Puppet::Rest`.

Depends on PR https://github.com/puppetlabs/puppet/pull/7825

### CODE REVIEW

**Comment by joshcooper** [comment]
jenkins please test this on redhat7-64a,windows2019-64a with servertests

**Comment by joshcooper** [comment]
jenkins please test this on redhat7-64a,windows2019-64a with servertests

**Comment by gimmyxd** [comment]
run a couple of modules acceptance suites:
```
~/Workspace/modules/tests/puppetlabs-powershell  origin/master !1 7m 9s
Finished in 4 minutes 20.8 seconds (files took 2 minutes 12.8 seconds to load)
29 examples, 0 failures


~/Workspace/modules/tests/puppetlabs-reboot  origin/master !1 5m 54s
Finished in 3 minutes 4.2 seconds (files took 2 minutes 27.1 seconds to load)
14 examples, 0 failures


~/Workspace/modules/tests/puppetlabs-iis  origin/master !1 26m 1s 
Pending: (Failures listed here are expected and do not affect your suite's status)
  1) iis_application when setting sslflags - blocked by MODULES-5561
     # No reason given
     # ./spec/acceptance/iis_application_spec.rb:243
Finished in 22 minutes 16 seconds (files took 3 minutes 15.9 seconds to load)
138 examples, 0 failures, 1 pending



~/Workspace/modules/puppetlabs-stdlib  origin/master 2m 5s
Finished in 1 minute 48.79 seconds (files took 13.26 seconds to load)
5 examples, 0 failures


~/Workspace/modules/puppetlabs-mysql  origin/master 12m 17s
Finished in 11 minutes 8 seconds (files took 1 minute 6.43 seconds to load)
91 examples, 0 failures


~/Workspace/modules/tests/puppetlabs-kubernetes  origin/master !1 39m 28s
Finished in 38 minutes 31 seconds (files took 54.12 seconds to load)
6 examples, 0 failures

```

File: lib/puppet/http/service/ca.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
+class Puppet::HTTP::Service::Ca < Puppet::HTTP::Service
+  HEADERS = { 'Accept' => 'text/plain' }.freeze
+
+  def get_certificate(name, ssl_context: nil)
```
Comment (gimmyxd) [inline lib/puppet/http/service/ca.rb]: Seems like Puppet::Rest::routes.get_certificates had an extra logging message `Puppet.info _("Downloaded certificate for %{name} from %{server}") % { name: name, server: ca.server }` - https://github.com/puppetlabs/puppet/blob/master/lib/puppet/rest/routes.rb#L43
This is causing the failures in `acceptance/tests/ssl/autosign_command.rb`
```    lax-vice.delivery.puppetlabs.net (lax-vice.delivery.puppetlabs.net) executed in 7.01 seconds
Beaker::DSL::Outcomes::FailTest:
Expected certificate to be autosigned.
Expected /Downloaded certificate for unfunny-vinegar.delivery.puppetlabs.net/ to match "\e[0;32mInfo: Resolving service 'ca' using Puppet::HTTP::Resolver::Settings\e[0m\n\e[0;32mInfo: Resolved service 'ca' to https://lax-vice.delivery.puppetlabs.net:8140/puppet-ca/v1\e[0m\n\e[0;32mInfo: Creating a new RSA SSL key for unfunny-vinegar.delivery.puppetlabs.net-autosign\e[0m\n\e[0;32mInfo: csr_attributes file loading from /etc/puppetlabs/puppet/csr_attributes.yaml\e[0m\n\e[0;32mInfo: Creating a new SSL certificate request for unfunny-vinegar.delivery.puppetlabs.net-autosign\e[0m\n\e[0;32mInfo: Certificate Request fingerprint (SHA256): 59:CD:6E:2D:15:0B:3F:51:3E:9A:23:26:76:33:70:63:36:1E:9E:ED:C2:16:02:85:A5:B7:93:3B:6A:77:64:75\e[0m\n\e[0;32mInfo: Using configured environment 'production'\e[0m\n\e[0;32mInfo: Retrieving pluginfacts\e[0m\n\e[0;32mInfo: Retrieving plugin\e[0m\n\e[0;32mInfo: Retrieving locales\e[0m\n\e[0;32mInfo: Caching catalog for unfunny-vinegar.delivery.puppetlabs.net-autosign\e[0m\n\e[0;32mInfo: Applying configuration version '1573566972'\e[0m\n\e[mNotice: Applied catalog in 0.01 seconds\e[0m\n".```

**Comment by joshcooper** [comment]
jenkins please test this on redhat7-64a,windows2019-64a with servertests

**Comment by Dorin-Pleava** [comment]
Did some manual tests regarding certificates in a redhat 7 PE and a windows 2012 node environment. Everything is working fine.
Also tested a little bit puppet agent upgrade and all is good.

**Comment by joshcooper** [comment]
jenkins please test this on redhat7-64a,windows2019-64a with servertests

**Pull request merged**